Prepare v1.4.1 release

dmcgowan opened the pull request:
Notes preview https://gist.github.com/dmcgowan/4aba9f29890a5576340568cb9480f439

AkihiroSuda commented:
Can we cherry-pick Go 1.15 https://github.com/containerd/containerd/pull/4050 ?

thaJeztah commented:
@AkihiroSuda any specific reason? I'd be more comfortable holding off for a bit. The bump to Go 1.15 was _just_ merged and I think we should give it some time there to see if there are no regressions (if possible). I realise go1.13 is now EOL, but there's not yet been security issues that affect containerd (that I'm aware of)

AkihiroSuda commented:
> I realise go1.13 is now EOL, but there's not yet been security issues that affect containerd (that I'm aware of)

:+1:

I'm ok to move https://github.com/containerd/containerd/pull/4050 to 1.4.2

thaJeztah commented on releases/v1.4.1.toml:
```diff
@@ -0,0 +1,35 @@
+* Fix devmapper device deletion on rollback [containerd/containerd#4437](https://github.com/containerd/containerd/pull/4437)
+* Seccomp default profile updates
+  * Add `openat2` (new in Linux 5.6) and `faccessat2` syscalls [containerd/containerd#4481](https://github.com/containerd/containerd/pull/4481)
+  * Move the syslog syscall to be gated by CAP_SYS_ADMIN or CAP_SYSLOG [containerd/containerd#4491](https://github.com/containerd/containerd/pull/4491)
```
```suggestion
  * Move the `syslog` syscall to be gated by `CAP_SYS_ADMIN` or `CAP_SYSLOG` [containerd/containerd#4491](https://github.com/containerd/containerd/pull/4491)
```

thaJeztah commented on releases/v1.4.1.toml:
```diff
@@ -0,0 +1,35 @@
+  * Add `openat2` (new in Linux 5.6) and `faccessat2` syscalls [containerd/containerd#4481](https://github.com/containerd/containerd/pull/4481)
+  * Move the syslog syscall to be gated by CAP_SYS_ADMIN or CAP_SYSLOG [containerd/containerd#4491](https://github.com/containerd/containerd/pull/4491)
+  * Add and update profile [containerd/containerd#4492](https://github.com/containerd/containerd/pull/4492)
+  * Add `io_uring_` syscalls [containerd/containerd#4493](https://github.com/containerd/containerd/pull/4493)
```
```suggestion
  * Allow `io_uring_` syscalls [containerd/containerd#4493](https://github.com/containerd/containerd/pull/4493)
```

thaJeztah reviewed (changes requested):
Left some suggestions

estesp reviewed (approved):
Agree with some of the suggested wording changes in comments.

Overall LGTM

thaJeztah commented:
do we need https://github.com/containerd/containerd/pull/4538 for v1.4.1? @fuweid ?

mikebrow commented on releases/v1.4.1.toml:
```diff
@@ -0,0 +1,35 @@
+### Notable Updates
+
+* Always consume shim logs to prevent logs in the shim from blocking [containerd/containerd#4546](https://github.com/containerd/containerd/pull/4546)
+* Update CNI Dependencies [containerd/containerd#4555](https://github.com/containerd/containerd/pull/4555)
```
nit add blurb:  `CNI v0.8.0 and CNI plugins v0.8.6 have been tested to work with this service update.`

dmcgowan replied:
Actually probably just remove this section as I don't see a specific reason it was backported in the first place. Just didn't quite make the cutoff. The dependencies changes should show up at the bottom, I'll see if I can get that fixed.

dmcgowan commented on releases/v1.4.1.toml:
```diff
@@ -0,0 +1,35 @@
+* Seccomp default profile updates
+  * Add `openat2` (new in Linux 5.6) and `faccessat2` syscalls [containerd/containerd#4481](https://github.com/containerd/containerd/pull/4481)
+  * Move the syslog syscall to be gated by CAP_SYS_ADMIN or CAP_SYSLOG [containerd/containerd#4491](https://github.com/containerd/containerd/pull/4491)
+  * Add and update profile [containerd/containerd#4492](https://github.com/containerd/containerd/pull/4492)
```
This section is already bordering lining on too much information, was thinking maybe collapsing even more and just mentioning seccomp default profile updates and having multiple links

thaJeztah replied:
Yes, that probably works as well

dmcgowan commented:
Updated, simplified the notable updates

> Can we cherry-pick Go 1.15

Let's wait a bit on this on, I think we should be very cautious about updating the major go version in our stable releases. We will need to in 1.4.x at some point though.

thaJeztah reviewed (approved):
LGTM

cpuguy83 reviewed (approved):
LGTM

estesp reviewed (approved):
LGTM

fuweid commented:
@thaJeztah 

> do we need #4538 for v1.4.1? @fuweid ?

sorry for late reply. just wait for next point release ~
